### PR TITLE
Login register screen

### PR DESCRIPTION
**Pull request by jacoblurie29**
Should work upon load (login screen should show instead of map view).

TEST EMAIL: nightlight@test.com
TEST PASSWORD: nightlight

### CODE REVIEW

File: src/config/firebaseConfig.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import firebase from 'firebase';
```
Comment (ethanratnofsky) [inline src/config/firebaseConfig.ts:1]:
```suggestion
import firebase from 'firebase';
import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword, signOut } from 'firebase/auth';
```

File: src/config/firebaseConfig.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  messagingSenderId: FIREBASE_MESSAGINGSENDERID,
+  appId: FIREBASE_APPID,
+  measurementId: FIREBASE_MEASUREMENTID,
+};
+
+// Check if firebase app exists before creating new instance
+if (!firebase.apps.length) {
+  const app = firebase.initializeApp(firebaseConfig);
+}
+
```
Comment (ethanratnofsky) [inline src/config/firebaseConfig.ts:27]:
```suggestion

const auth = getAuth();

```

File: src/config/firebaseConfig.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * @param email valid string email address
+ * @param password valid password for user account
+ *
+ * @returns object that contains a user authentication token as well as other related information
+ */
+export const handleSignUp = async (email: string, password: string) => {
+  await firebase
+    .auth()
+    .createUserWithEmailAndPassword(email, password)
```
Comment (ethanratnofsky) [inline src/config/firebaseConfig.ts:39]:
```suggestion
  await createUserWithEmailAndPassword(email, password)
```

File: src/config/firebaseConfig.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    .then(() => {
+      console.log(firebase.auth().currentUser);
```
Comment (ethanratnofsky) [inline src/config/firebaseConfig.ts:41]:
```suggestion
      console.log(auth.currentUser);
```

File: src/config/firebaseConfig.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * @param email valid string email address
+ * @param password valid password for user account
+ *
+ * @returns object that contains a user authentication token as well as other related information
+ */
+export const handleLogin = async (email: string, password: string) => {
+  await firebase
+    .auth()
+    .signInWithEmailAndPassword(email, password)
```
Comment (ethanratnofsky) [inline src/config/firebaseConfig.ts:60]:
```suggestion
  await signInWithEmailAndPassword(email, password)
```

File: src/config/firebaseConfig.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    .then(() => {
+      console.log(firebase.auth().currentUser);
```
Comment (ethanratnofsky) [inline src/config/firebaseConfig.ts:62]:
```suggestion
      console.log(auth.currentUser);
```

File: src/config/firebaseConfig.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    });
+};
+
+/**
+ * Remove the existing user from the firebase application
+ */
+export const handleSignOut = async () => {
+  await firebase
+    .auth()
+    .signOut()
```
Comment (ethanratnofsky) [inline src/config/firebaseConfig.ts:76]:
```suggestion
  await signOut()
```

File: src/config/firebaseConfig.ts [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    .then(() => {
+      console.log(firebase.auth().currentUser);
```
Comment (ethanratnofsky) [inline src/config/firebaseConfig.ts:78]:
```suggestion
      console.log(auth.currentUser);
```

File: src/screens/login/LoginScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+                  placeholder='Password'
+                  secureTextEntry={isPasswordHidden}
+                />
+              )}
+              name='password'
+              rules={{ required: true }}
+            />
+            <Pressable
+              style={LoginScreenStyles.viewPasswordButton}
+              onPress={() => setIsPasswordHidden(!isPasswordHidden)}>
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:93]:
```suggestion
              onPress={toggleIsPasswordHidden}>
```

File: src/screens/login/LoginScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+import {
+  handleLogin,
+  handleSignOut,
+} from '@nightlight/src/config/firebaseConfig';
+import { Controller, useForm } from 'react-hook-form';
+
+const LoginScreen = ({ setIsLogin }: LoginCardProps) => {
+  // state for password hiding (passwords are hidden by default)
+  const [isPasswordHidden, setIsPasswordHidden] = useState(true);
+
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:26]:
```suggestion

  const toggleIsPasswordHidden = () => setIsPasswordHidden(prev => !prev);

```

File: src/screens/login/LoginScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+        password: '',
+      });
+    });
+  };
+
+  return (
+    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
+      <SafeAreaView style={LoginScreenStyles.container}>
+        <View style={LoginScreenStyles.logoDot}></View>
+        <View style={LoginScreenStyles.logoBody}></View>
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:55]:
```suggestion
        <View style={LoginScreenStyles.logoDot} />
        <View style={LoginScreenStyles.logoBody} />
```

File: src/screens/login/LoginScreen.tsx [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+  Text,
+  SafeAreaView,
+  TextInput,
+  Pressable,
+  Image,
+  Linking,
+  TouchableWithoutFeedback,
+  Keyboard,
+} from 'react-native';
+import LoginScreenStyles from './LoginScreen.styles';
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:12]:
```suggestion
import LoginScreenStyles from '@nightlight/screens/login/LoginScreen.styles';
```

File: src/screens/login/LoginScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+          </View>
+
+          <View style={LoginScreenStyles.forgotPasswordContainer}>
+            <Text style={LoginScreenStyles.forgotPassword}>
+              Forgot password?
+            </Text>
+          </View>
+          <Pressable
+            style={LoginScreenStyles.signInButton}
+            onPress={handleSubmit(onSubmit)}>
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:113]:
```suggestion
            onPress={() => handleSubmit(onSubmit)}>
```
not sure abt this? does this behave differently with hook forms?

File: src/screens/login/LoginScreen.tsx [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+import LoginScreenStyles from './LoginScreen.styles';
+import { Entypo } from '@expo/vector-icons';
+import { COLORS } from '@nightlight/src/global.styles';
+import { useState } from 'react';
+import { LoginCardProps, LoginFormData } from '@nightlight/src/types';
+import {
+  handleLogin,
+  handleSignOut,
+} from '@nightlight/src/config/firebaseConfig';
+import { Controller, useForm } from 'react-hook-form';
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:21]:
```suggestion
import { Controller, useForm } from 'react-hook-form';
import GoogleIcon from '../../../assets/googleIcon.png';
```

Comment (ethanratnofsky) [reply src/screens/login/LoginScreen.tsx:21]: can you use the '@nightlight' alias here?

File: src/screens/login/LoginScreen.tsx [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+            style={LoginScreenStyles.signInButton}
+            onPress={handleSubmit(onSubmit)}>
+            <Text style={LoginScreenStyles.signInButtonText}>Sign In</Text>
+          </Pressable>
+          <Text style={LoginScreenStyles.continueWithText}>
+            Or continue with
+          </Text>
+          <Pressable style={LoginScreenStyles.googleSignInButton}>
+            <Image
+              source={require('../../../assets/googleIcon.png')}
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:121]:
```suggestion
              source={GoogleIcon}
```

File: src/screens/login/LoginScreen.tsx [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+              style={LoginScreenStyles.googleIcon}
+            />
+            <Text style={LoginScreenStyles.googleSignInButtonText}>
+              Sign in with Google
+            </Text>
+          </Pressable>
+          <View>
+            <Text style={LoginScreenStyles.notMemberText}>
+              Not a member?&nbsp;&nbsp;
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:130]:
```suggestion
              Not a member?
```
idk what those chars are

File: src/screens/login/LoginScreen.tsx [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+              <Text
+                onPress={() => setIsLogin(false)}
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:132]:
```suggestion
                onPress={handleRegisterLinkPress}
```

File: src/screens/login/LoginScreen.tsx [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+  const onSubmit = (data: any) => {
+    console.log('handling login', data);
+    handleLogin(data.email, data.password).then(() => {
+      reset({
+        email: '',
+        password: '',
+      });
+    });
+  };
+
```
Comment (ethanratnofsky) [inline src/screens/login/LoginScreen.tsx:50]:
```suggestion

  const handleRegisterLinkPress = () => setLogin(false);

```

File: src/screens/profile/ProfileScreen.tsx [hunk 19]
```diff
@@ -0,0 +1,19 @@
+import { Pressable, SafeAreaView, Text } from 'react-native';
+import { handleSignOut } from '../../config/firebaseConfig';
```
Comment (ethanratnofsky) [inline src/screens/profile/ProfileScreen.tsx:2]:
```suggestion
import { handleSignOut } from '@nightlight/src/config/firebaseConfig';
```

File: src/screens/profile/ProfileScreen.tsx [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+import ProfileScreenStyles from './ProfileScreen.styles';
```
Comment (ethanratnofsky) [inline src/screens/profile/ProfileScreen.tsx:3]:
```suggestion
import ProfileScreenStyles from '@nightlight/screens/profile/ProfileScreen.styles';
```

File: src/screens/profile/ProfileScreen.tsx [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+
+const ProfileScreen = () => {
+  return (
+    <SafeAreaView>
+      <Pressable style={ProfileScreenStyles.logOutButton}>
+        <Text
+          style={ProfileScreenStyles.logOutButtonText}
+          onPress={() => handleSignOut()}>
```
Comment (ethanratnofsky) [inline src/screens/profile/ProfileScreen.tsx:11]:
```suggestion
          onPress={handleSignOut}>
```

File: src/screens/register/RegisterScreen.tsx [hunk 22]
```diff
@@ -0,0 +1,186 @@
...
+  View,
+  Text,
+  SafeAreaView,
+  TextInput,
+  Pressable,
+  Image,
+  TouchableWithoutFeedback,
+  Keyboard,
+} from 'react-native';
+import RegisterScreenStyles from './RegisterScreen.styles';
```
Comment (ethanratnofsky) [inline src/screens/register/RegisterScreen.tsx:11]:
```suggestion
import RegisterScreenStyles from '@nightlight/screens/register/RegisterScreen.styles';
```

File: src/screens/register/RegisterScreen.tsx [hunk 23]
```diff
@@ -0,0 +1,186 @@
...
+              value={value}
+            />
+          )}
+          name='phone'
+          rules={{ required: true }}
+        />
+        <Pressable style={RegisterScreenStyles.signInButton}>
+          <Text
+            style={RegisterScreenStyles.signInButtonText}
+            onPress={handleSubmit(onSubmit)}>
```
Comment (ethanratnofsky) [inline src/screens/register/RegisterScreen.tsx:156]: see previous related comment

File: src/screens/register/RegisterScreen.tsx [hunk 24]
```diff
@@ -0,0 +1,186 @@
...
+                onChangeText={value => onChange(value)}
+                value={value}
+              />
+            )}
+            name='password'
+            rules={{ required: true }}
+          />
+          <Pressable
+            style={RegisterScreenStyles.viewPasswordButton}
+            onPress={() => setIsPasswordHidden(!isPasswordHidden)}>
```
Comment (ethanratnofsky) [inline src/screens/register/RegisterScreen.tsx:92]: see comment on LoginScreen.tsx

File: src/screens/register/RegisterScreen.tsx [hunk 25]
```diff
@@ -0,0 +1,186 @@
...
+            Sign Up
+          </Text>
+        </Pressable>
+        {/* End form */}
+        <Text style={RegisterScreenStyles.continueWithText}>
+          Or continue with
+        </Text>
+        <Pressable style={RegisterScreenStyles.googleSignInButton}>
+          <Image
+            source={require('../../../assets/googleIcon.png')}
```
Comment (ethanratnofsky) [inline src/screens/register/RegisterScreen.tsx:166]: see comment on LoginScreen.tsx

File: src/screens/register/RegisterScreen.tsx [hunk 26]
```diff
@@ -0,0 +1,186 @@
...
+            style={RegisterScreenStyles.hasAccountLink}>
+            Login now
+          </Text>
+        </Text>
+      </SafeAreaView>
+    </TouchableWithoutFeedback>
+  );
+};
+
+export default RegisterScreen;
```
Comment (ethanratnofsky) [inline src/screens/register/RegisterScreen.tsx:186]: there seems to be a lot of reused code from LoginScreen.tsx......................refactor to use conditional rendering in one screen :)

File: src/types.ts [hunk 27]
```diff
@@ -1,11 +1,25 @@
...
+  password: string;
+  confirmPassword: string;
+  phone: string;
+};
+
+export type LoginFormData = {
+  email: string;
+  password: string;
+};
+
```
Comment (ethanratnofsky) [inline src/types.ts:14]:
```suggestion
export interface RegisterFormData {
  email: string;
  password: string;
  confirmPassword: string;
  phone: string;
};

export interface LoginFormData {
  email: string;
  password: string;
};

```

**Review by ethanratnofsky — CHANGES_REQUESTED** [review]

**Comment by ethanratnofsky** [comment]
in addition to changes requested above, pls merge updated main into this branch :)

File: .env.template [hunk 28]
```diff
@@ -1,2 +0,0 @@
-# Map Box API Key
-MAPBOX_API_KEY =
```
[the remark this answers is not in the document]
Comment (jacoblurie29) [reply .env.template:2]: Hehe

**Pull request closed without merging**